osu.Framework/Lists/SortedList.cs

```diff
@@ -112,7 +112,7 @@ public int RemoveAll(Predicate<T> match)
         public int IndexOf(T value)
         {
             int index = list.BinarySearch(value, Comparer);
-            return index >= 0 && list[index].Equals(value) ? index : -1;
+            return index >= 0 ? index : -1;
```

Comment:
```suggestion
            return index;
```

Preserves the functionality of binarysearch